Testscripts/Windows/PERF-LKP.ps1

```diff
@@ -0,0 +1,248 @@
+
+    try {
+
+        $LogPath = Join-Path $LogDir\$LKP_LOG_DIR $testName
+        if (-not (Test-Path -Path $LogPath)) {
+            Write-LogErr "EXCEPTION : $LKP_LOG_DIR doesn't exist"
+            return $resultAborted
+        }
+
+        $items = Get-ChildItem $LogPath
+        $items | ForEach-Object {
+            $resultPath = Join-Path $LogPath $_
+            $resultFile = Join-Path $resultPath "${testName}.json"
+            if (-not (Test-Path -Path $resultFile)) {
+                Write-LogErr "EXCEPTION : $resultFile doesn't exist"
+                return $resultPass
+            }
+
+            $metaData = "{0,-30}" -f $_
+            $result = $null
+            $item = "{0,-30} {1,30}" -f $metaData, $result
+            Add-Content -Value $item -Path "$LogDir\$LKP_RESULTS"
+            $CurrentTestResult.TestSummary += New-ResultSummary -testResult $result -metaData $metaData `
+                    -checkValues "PASS,FAIL,SKIP" -testName $CurrentTestData.testName
+
+            (Get-Content $resultFile) | ConvertFrom-Json > tmp.txt
+            $content = Get-Content ".\tmp.txt"
+            $content | ForEach-Object {
+                $isBlank = [string]::IsNullOrWhiteSpace($_)
+                if (-not $isBlank) {
+                    $metaData = "    $($_.split(':')[0])"
+                    $result = $_.split(":")[1].split("{}")[1]
+
+                    $item = "{0,-30} {1,30}" -f $metaData, $result
+                    Add-Content -Value $item -Path "$LogDir\$LKP_RESULTS"
+                    $CurrentTestResult.TestSummary += New-ResultSummary -testResult $result -metaData $metaData `
+                            -checkValues "PASS,FAIL,SKIP" -testName $CurrentTestData.testName
+                }
+            }
+            Remove-Item ".\tmp.txt"
+        }
+        return $resultPass
+
+    } catch {
+        Write-LogErr "Getting test summary results failed"
+        Remove-Item ".\tmp.txt"
+        $errorMessage =  $_.Exception.Message
+        $errorLine = $_.InvocationInfo.ScriptLineNumber
+        Write-LogErr "EXCEPTION : $errorMessage at line: $errorLine"
+        return $resultAborted
+    }
+}
+function Get-HorizontalFormat-TestResult ($currentTestResult, $currentTestData, $testName) {
+    try {
+        $isResultHead = $true
+        $metaDataTitle = "{0,-30}" -f "Jobs"
+
+        $LogPath = Join-Path $LogDir\$LKP_LOG_DIR $testName
+        if (-not (Test-Path -Path $LogPath)) {
+            Write-LogErr "EXCEPTION : $LKP_LOG_DIR\$testName doesn't exist"
+            return $resultAborted
+        }
+
+        $items = Get-ChildItem $LogPath
+        $items | ForEach-Object {
+            $resultPath = Join-Path $LogPath $_
+            $resultFile = Join-Path $resultPath "${testName}.json"
+            if (-not (Test-Path -Path $resultFile)) {
+                Write-LogErr "EXCEPTION : $resultFile doesn't exist"
+                return $resultPass
```

Comment:
not exist, then test pass?

Reply:
Yeah, for PERF-LKP-MICROBENCHMARK test case, there are a few benchmarks. We don't expect the test exit if one of them has no result file.